mlir/lib/Dialect/Linalg/Transforms/Vectorization.cpp

```diff
@@ -1609,6 +1610,19 @@ static LogicalResult reductionPreconditions(LinalgOp op) {
 }
 
 static LogicalResult vectorizeDynamicLinalgOpPrecondition(linalg::LinalgOp op) {
+  // Support dynamic shapes in 1D depthwise convolution, but only in the
+  // _channel_ dimension. That's exclusively to support scalable vectorisation.
+  if (auto conv = dyn_cast<linalg::DepthwiseConv1DNwcWcOp>(op.getOperation())) {
+    auto lhsShaped = op.getDpsInputOperand(0)->get();
+    ArrayRef<int64_t> lhsShape =
+        dyn_cast<ShapedType>(lhsShaped.getType()).getShape();
+    auto shapeWithoutCh = lhsShape.drop_back(1);
+    if (ShapedType::isDynamicShape(shapeWithoutCh))
+      return failure();
+
+    return success();
```

Comment:
Not sure I follow - we need to verify that all dimensions except for the channel dim are static. Checking the channel dim alone wouldn't be sufficient. Perhaps share a snippet so that I know what you meant?

Reply:
ah ok, I was suggesting
```suggestion
    if (ShapedType::isDynamicShape(lhsShape.back()))
      return success();

    return failure();
```
but as you say that's not checking if the other dims are static. Perhaps worth adding a negative test with dynamic dims other than channel to check nothing happens.